### PR TITLE
Chore/codebase improvements

### LINKED ISSUE #89: Improve build process with minification
Currently, we're relying on tsc to compile the code which in turn is generating un-minified code. 
we could bundle the code with Rollup with `rollup-plugin-uglify` to get a smaller bundle size

### LINKED ISSUE #137: Jest complains about createStyled
Hello!

I'm trying to test a component with Jest and React Testing Library and I got this error:

image

I'm not sure if it's something I can fix on my end or is something related to stitches

### LINKED ISSUE #147: Improve release flow
I think we should:
- build a single `dist` folder for each package, instead of a `lib` and `es`
- set up prettier
- make sure prettier is run before release (right now the releases fail, right at the end, if there's a prettier error. very frustrating)
- remove package-lock and stick with yarn
- see if it can be faster by removing unnecessary steps, if any
- rename `packages/css` to `packages/core` so its consistent with the npm package name
- rename `packages/styled` to `packages/react` so its consistent with the npm package name
- review all steps to make sure we're not doing anything we don't need to

### PR DESCRIPTION
**Pull request by hadihallak**
**What's inside:**
- Improve release flow and build (building both packages take around 5sec on my machine)
- Improve dev experience ( now you will be able to work on on both the react and core packages simultaneously without linking or doing anything fancy thanks to preconstruct)
- moved the bundles to dist folder and generate both dev and production build (also thanks to preconstruct 👌 )
- match folder names to package names and fix typing issue related to it
- Add testing foundation to the react package and initial tests to test basic functionality such as variants, compound variants, and the css prop
- Added Custom jest serializer to test styled components in a meaningful way (this still needs some work but it now allows us to not only see the rendered class names but also the CSS rules in the order of injection)
- add pre-commit prettier and linting rules and run it on the codebase to keep things consistent 
- update monorepo to use yarn instead of npm 
- Fixes components not being testable in jest #137 

Fixes #147
Fixes #89

### CODE REVIEW

**Comment by hadihallak** [comment]
@peduarte FIY

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Not quite ready :)
 - Christian Alfoni ([@christianalfoni](https://twitter.com/christianalfoni)) - Codesandbox, Cerebral
 - Pedro Duarte ([@peduarte](https://twitter.com/peduarte)) - [Modulz](https://modulz.app)
 - Jenna Smith ([@jjenzz](https://twitter.com/jjenzz)) - [Modulz](https://modulz.app)
-- Abdul Hallak ([@hadi_hlk](https://twitter.com/hadi_hlk)) - [Modulz](https://modulz.app)
+- Abdulhadi Alhallak ([@hadi_hlk](https://twitter.com/hadi_hlk)) - [Modulz](https://modulz.app)
```
Comment (peduarte) [inline README.md:24]: 😄

File: package.json [hunk 2]
```diff
@@ -23,8 +30,21 @@
...
   "dependencies": {
-    "@stitches/core": "file:packages/css",
-    "@stitches/react": "file:packages/styled"
+    "@babel/core": "^7.11.5",
+    "@babel/preset-env": "^7.11.5",
+    "@babel/preset-typescript": "^7.10.4",
+    "@preconstruct/cli": "^1.1.27",
+    "react": "^16.13.1"
+  },
+  "preconstruct": {
```
Comment (peduarte) [inline package.json:45]: whaqts preconstruct?

**Comment by peduarte** [comment]
Great stuff, thanks for that. I'll do a bunch of testing today.

You didn't change the tslint config. do you think we should? I remember seeing loads of tslint ignores throughout...

**Comment by peduarte** [comment]
@hadihallak I found an issue with the tests after building.

To replicate:
- run yarn test :white_check_mark: 
- run yarn build :white_check_mark:
- run yarn test :x:

File: package.json (hunk 2 above)
Comment (hadihallak) [reply package.json:45]: https://preconstruct.tools/
it makes life a lot easier when working with libraries in monorepos

**Comment by hadihallak** [comment]
> You didn't change the tslint config. do you think we should? I remember seeing loads of tslint ignores throughout...

will check this today

**Pull request merged**